lib/parsers/ltx.js

```diff
@@ -42,8 +44,17 @@ var SaxLtx = module.exports = function SaxLtx () {
     }
   }
 
+  this._setLast = function (last) {
+    isLast = last
+  }
+
   this.write = function (data) {
-    if (typeof data !== 'string') {
+    if (Buffer.isBuffer(data)) {
```

Comment:
`Buffer` is undefined in browser

`TextDecoder` can also read `ArrayBuffer` and `ArrayBufferView`

Also, please keep current behavior if `TextDecoder` isn't available

something like:

```js
function decodable(data) {
  return typeof TextDecoder !== undefined && is array buffer || is array buffer view
}


if (typeof data !== 'string') {
  if (isDecodable(data)) {
    ... create decoder if it does not exist
    data = decoder.decode(data)
  } else {
    data = data.toString()
  }
}
```


for reference / inspiration https://github.com/isaacs/sax-js/blob/master/lib/sax.js#L228

Reply:
Came up with this:
this._isDecodable = function (data) {
    return data && typeof TextDecoder !== 'undefined' &&
      (
        (typeof Buffer === 'function' && Buffer.isBuffer(data)) ||
        (typeof ArrayBuffer === 'function' && (data instanceof ArrayBuffer || Object.prototype.toString.call(data) === '[object ArrayBuffer]' || ArrayBuffer.isView(data)))
      )
  }